spec/models/game_spec.rb

```diff
@@ -0,0 +1,137 @@
+describe Game do
+  subject { described_class.new(frame_class: frame_class, score_board: score_board) }
+
+  let(:frame) { instance_double(Frame, :frame, add_roll: nil, over?: false) }
+  let(:frame_class) { class_double(Frame, :frame_class, new: frame) }
+  let(:score_board) { class_double(ScoreBoard, :score_board, total_score: nil) }
+
+  describe '#score' do
+    it 'calls on score_board' do
+      expect(score_board).to receive(:total_score).once
+      subject.score
+    end
+  end
+
+  describe '#current_frame' do
+    it 'holds an instance of frame' do
+      expect(subject.current_frame).to be frame
```

Comment:
This test is testing state not behaviour. Your test should not fix your implementation rigidly, and so they should only be testing inputs and outputs of your class.
outputs include how you class calls other classes and what they call them with (which you do). But the internal state should stay hidden.

Reply:
cool, that makes sense. similar to how there's no need to test private methods - we only need to test for desired behaviour of the user?

In this case would you say we could remove the test for this method altogether?

Reply:
Yes, absolutely. Also, maybe this is not a public method?